ts/Stock/Navigator/Navigator.ts

```diff
@@ -1218,6 +1291,20 @@ class Navigator {
             }
         }
     }
+    /**
+     * Calculate the navigator xAxis offset and height
+     *
+     * @private
+     */
+    public getXAxisOffsetAndHeight(): DeepPartial<AxisOptions> {
+        const chart = this.chart;
+        return {
+            offsets: chart.inverted ?
+                [this.scrollButtonSize, 0, -this.scrollButtonSize, 0] :
+                [0, -this.scrollButtonSize, 0, this.scrollButtonSize],
+            [chart.inverted ? 'width' : 'height']: this.height
```

Comment:
Small mistake: sometimes you return `width`, sometimes `height`, but in code later you always take `height` (which might be in an inverted chart, returning undefined). 

1. Please narrow down the type. `DeepPartial<AxisOptions>` is too broad and TS didn't tell you that mistake
2. Consider resigning from width/height. It seems in places where the method is called, the `height` variable is defined anyway (different scope though)